core/src/mender-client.c

```diff
@@ -1230,6 +1249,7 @@ mender_client_update_work_function(void) {
             ret = MENDER_FAIL;
             goto END;
         }
+        cJSON_AddNumberToObject(mender_client_deployment_data, "version", (double)MENDER_CLIENT_DEPLOYMENT_DATA_VERSION);
         cJSON_AddStringToObject(mender_client_deployment_data, "id", deployment->id);
         cJSON_AddStringToObject(mender_client_deployment_data, "artifact_name", deployment->artifact_name);
         cJSON_AddArrayToObject(mender_client_deployment_data, "types");
```

Comment:
Only difference is that provides is stored as a string with the
```C
/* ASCII unit separator */
#define MENDER_KEY_VALUE_DELIMITER "\x1F"
/* ASCII record separator */
#define MENDER_KEY_VALUE_SEPARATOR "\x1E"
```
separators/delimiters.

Maybe we should store it as an array instead? I could fix that as a part of this PR.

Reply:
Sorry, I think my point was not clear,

I meant to just add `cJSON_AddArrayToObject(mender_client_deployment_data, "provides");` here, even if the contents are added only later (same as for `types`).

Reply:
I think we are misunderstanding each other. The JSON object currently looks like this
```json
{
  "id": "1fcbf9e6-4786-4358-8ec1-2c4107068d6a",
  "artifact_name": "test123",
  "types": ["rootfs-image"],
  "provides": "artifact_name\u001ftest123\u001erootfs-image.version\u001ftest123\u001erootfs-image.checksum\u001fa676d107f1137634cd29ef814878c5e53eb0ecfe095f16da4e0cfaf04eecf5b6\u001e"
}
```

Provides is not an array. Hence, it does not make sense to initialize it as an array, unless we change the code to store it as an array instead.

Reply:
Oh, got it! Sorry.

And what if we initialize it to empty string? Is that a bad idea in the sense that it could indicate basically that the Artifact provides nothing?

On the high level what I am trying to advocate for is to create the object in one place. But it might not make sense for this field, as `""` is a valid value.